ocaml/typing/printpat.ml

```diff
@@ -24,22 +24,27 @@ let is_cons = function
 | {cstr_name = "::"} -> true
 | _ -> false
 
+let fmt_unboxed_const_str s =
+  match String.split_on_char '-' s with
+  | [""; s] -> "-#" ^ s
+  | [s] -> "#" ^ s
+  | _ -> assert false
```

Comment:
This is not expressly related to this PR, but I don't see analogous code in the parsetree's `pprintast.ml`. That makes me think that `-dsource` won't appropriately round trip. Light testing with the top-level seems to confirm.

Perhaps a good regression test would be to add an example to `ocaml/testsuite/tests/parsetree/source_jane_street.ml`. That's become a bit more comprehensive since our first go at unboxed literals.